cmd/notary/keys.go

```diff
@@ -599,7 +575,11 @@ func (k *keyCommander) keyPassphraseChange(cmd *cobra.Command, args []string) er
 	// unlocking passphrase and reusing that.
 	passChangeRetriever := k.getRetriever()
 	keyStore, err := trustmanager.NewKeyFileStore(config.GetString("trust_dir"), passChangeRetriever)
-	err = keyStore.AddKey(filepath.Join(keyGUN, keyID), role, privKey)
+	keyInfo, err := cs.GetKeyInfo(keyID)
+	if err != nil {
+		return err
+	}
+	err = keyStore.AddKey(privKey, keyInfo)
```

Comment:
Should we ensure we're adding the key back to the same keystore we found it in? Easiest way to do that would be to iterate keystores, create a CryptoService with only one key store at a time and attempt to Get/Add via the CryptoService.